code/lib/cli/src/repro-generators/scripts.ts

```diff
@@ -287,8 +291,24 @@ export const createAndInit = async (
   logger.info(`🏃 Starting for ${name} ${version}`);
   logger.log();
 
-  await doTask(generate, { ...options, cwd: options.creationPath });
+  try {
+    if (!options.preferGitTemplate) {
+      throw new Error('using local generators instead..');
+    }
+    // try downloading from git repro first
+    await doTask(generate, {
+      ...options,
+      generator: options.gitRepoGenerator,
+      cwd: options.creationPath,
+    });
+  } catch (err) {
+    // fallback to local generators
+    logger.info(`⏩ Skipping git template and using local generator instead`);
+    await doTask(generate, { ...options, cwd: options.creationPath });
+  }
```

Comment:
I find this control flow via `throw` to be pretty confusing. My suggestion would be something like:

```ts
let { preferGitTemplate } = options;
try {
  if (preferGitTemplate) await doTask(...);
} catch(err) {
  logger.info(...);
  preferGitTemplate = false;
}

if (!preferGitTemplate) await doTask(...);
```

Reply:
Or even better restructure to something like

```ts
try {
  if (options.preferGitTemplate) return doTask(...)
} catch (err) {
   logger.info(...);
}
return doTask(...);
```